test/t8_IO/t8_gtest_vtk_writer.cxx

```diff
@@ -114,6 +116,24 @@ use_c_interface<t8_cmesh_t> (const t8_cmesh_t grid, const char *fileprefix, [[ma
 #endif
 }
 
+void
+vtk_writer_test_fill_data (const t8_locidx_t cells_to_write_count, std::vector<double> &scalar_data,
+                           std::vector<double> &vector_data)
+{
+  // Fill scalar data vector with entries 0,1/10,...,(N-1)/10
+  //    vector[n] = (n/10.)
+  // Fill vector data vector with entries (0, 0, 42), (0.1,-0.1,42), ...
+  //    vector[n] = (n/10.,-n/10., 42)
+  for (t8_locidx_t icell = 0; icell < cells_to_write_count; ++icell) {
```

Comment:
```suggestion
 std::iota(scalar_data.begin(), scalar_data.end(), 0);
  std::transform(scalar_data.begin(), scalar_data.end(), scalar_data.begin(), [](double val) { return val / 10.; });

  std::generate(vector_data.begin(), vector_data.end(), [n = 0]() mutable {
    double scalar_value = (n / 10.);
    double vector_values[3] = { scalar_value, -scalar_value, 42. };
    return vector_values[n++ % 3];
  });
```

we could replace the loop by stl commands then. It has the drawback of looping three times over the arrray, though, so I would be okay with just removing the allocation from the loop too.

Reply:
Yes good point. I used the generator now.